scene/gui/tree.cpp

```diff
@@ -3893,16 +3889,16 @@ TreeItem *Tree::get_last_item() const {
 	return last;
 }
 
-void Tree::item_edited(int p_column, TreeItem *p_item, bool p_lmb) {
+void Tree::item_edited(int p_column, TreeItem *p_item, MouseButton p_mouse_index) {
 	edited_item = p_item;
 	edited_col = p_column;
 	if (p_item != nullptr && p_column >= 0 && p_column < p_item->cells.size()) {
 		edited_item->cells.write[p_column].dirty = true;
 	}
-	if (p_lmb) {
+	if (p_mouse_index == MouseButton::NONE) {
 		emit_signal(SNAME("item_edited"));
 	} else {
-		emit_signal(SNAME("item_rmb_edited"));
+		emit_signal(SNAME("item_edit_mouse_clicked"), p_mouse_index);
 	}
```

Comment:
From the old code it seems like both signals were for mouse clicks, `p_lmb` was just for the left click. It's ok to divide it like that, but the method doesn't properly get the mouse buttons (e.g. clicking a checkable item doesn't emit the mouse clicked signal, because it calls `item_edited()` without the last argument).

Reply:
The problem with `item_edited` is that it's not always called from a mouse event, but may come from a keyboard event or even an event that was fired outside Tree (for both, `Tree::_text_editor_submit` is an example). Maybe I can propagate the mouse button when it's known, but it won't be really consistent or predictable (because it depends on how the user use the Tree).

Reply:
No I mean e.g. here:
```C++

				if (force_edit_checkbox_only_on_checkbox) {
					if (x < cache.checked->get_width()) {
						p_item->set_checked(col, !c.checked);
						item_edited(col, p_item);
					}
				} else {
					p_item->set_checked(col, !c.checked);
					item_edited(col, p_item);
				}
```
This is done in response to mouse click, but you aren't passing the mouse button.

Reply:
I did a quick pass where `item_edited` is used and added the mouse button where possible.